source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -469,6 +481,41 @@ ShaderGraphPtr ShaderGraph::create(const string& name, ElementPtr element, Shade
             }
         }
 
+        // Add shareRef nodedef paths
+        const vector<InputPtr> nodeInputs = nodeDef->getChildrenOfType<Input>();
+        const string& nodePath = shaderRef->getNamePath();
+        for (const ValueElementPtr& nodeInput : nodeInputs)
+        {
+            const string& inputName = nodeInput->getName();
+            const string path = nodePath + NAME_PATH_SEPARATOR + inputName;
```

Comment:
I wonder if the path in this case should be the path to the BindInput/BindParam on the shaderref instead? Because on client side those are the actual elements that gets changed when a value changes.

Reply:
I think this is what it does as it get's the input/params from the shaderref and then uses:
shaderRef_name_<input/param> name?

Reply:
Ah you're right, the BindInput name matches the Input name.